### PR TITLE
✨ Implement ignoreFields in server side apply

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes https://github.com/open-cluster-management-io/ocm/issues/690

### CODE REVIEW

**Comment by zhujian7** [comment]
LGTM

File: pkg/work/spoke/apply/server_side_apply_test.go [hunk 1]
```diff
@@ -279,3 +294,251 @@ func newManifestWork() *unstructured.Unstructured {
...
+				removeFieldByJSONPath(desired.UnstructuredContent(), jsonPath, logger)
+			}
+			if !equality.Semantic.DeepEqual(c.exp, desired) {
+				t.Errorf("expected %v, got %v", c.exp, desired)
+			}
+		})
+	}
+}
+
+func TestServerSideApplyWithIgnoreFields(t *testing.T) {
```
Comment (haoqing0110) [inline pkg/work/spoke/apply/server_side_apply_test.go:426]: Does it make sense if ignorefields is .metadata.xxx? For example, what if ignorefields is .metadata.annotations and type is IgnoreOnSpokePresent, will the existing annotations be kept on the spoke or be override by object-hash?

Comment (qiujian16) [reply pkg/work/spoke/apply/server_side_apply_test.go:426]: it will be kept. We compute the hash for the whole object including metadata.

**Pull request merged**